### PR TITLE
2.1.0

### PR DESCRIPTION
**Pull request by integrii**
- The new CheckReaper (included in Helm chart specs and flat files) will now reap checks sitting around in the state of `Completed` but leave checks that have failed around longer for inspection.   @joshulyne 
- New namespace filtering on the JSON status page output with the GET variable `namespace`.  Use commas for multiple namespaces.  Example: `?namespace=kuberhealthy,kube-system` @jonnydawg 
- Check run duration has been added to `khstate` information and the status page. @joshulyne 
- Many bug fixes as listed [in the 2.1.0 milestone](https://github.com/Comcast/kuberhealthy/milestone/3?closed=1)

### CODE REVIEW

**Comment by jonnydawg** [comment]
Addressed my previous comment with that previous commit.

File: cmd/check-reaper/check-reaper.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+spec:
+  schedule: "*/3 * * * *"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+            - name: check-reaper
+              image: quay.io/comcast/check-reaper:1.0.0
+              imagePullPolicy: Always
```
Comment (joshulyne) [inline cmd/check-reaper/check-reaper.yaml]: change imagepullpolicy to IfNotPresent?

Comment (joshulyne) [reply cmd/check-reaper/check-reaper.yaml]: this runs every 3 mins, so best not to pull it so often

File: cmd/deployment-check/deployment.go [hunk 2]
```diff
@@ -326,50 +329,74 @@ func createContainerConfig(imageURL string) corev1.Container {
...
+			// Check if we have timed out.
+			select {
+			case <-ctx.Done():
+				deleteChan <- fmt.Errorf("timed out while waiting for deployment to delete")
+			default:
+				log.Debugln("Delete deployment and wait has not yet timed out.")
+			}
+
+			// Wait between checks.
+			log.Debugln("Waiting 5 secodns before trying again.")
```
Comment (joshulyne) [inline cmd/deployment-check/deployment.go]: typo "seconds" -- but thats just me being nitpicky :)

File: deploy/helm/kuberhealthy/templates/check-reaper.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+spec:
+  schedule: "*/3 * * * *"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+            - name: check-reaper
+              image: quay.io/comcast/check-reaper:1.0.0
+              imagePullPolicy: Always
```
Comment (joshulyne) [inline deploy/helm/kuberhealthy/templates/check-reaper.yaml]: Set to IfNotPresent.

File: deploy/kuberhealthy-prometheus-operator.yaml [hunk 4]
```diff
@@ -313,11 +382,32 @@ spec:
...
+spec:
+  schedule: "*/3 * * * *"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+            - name: check-reaper
+              image: quay.io/comcast/check-reaper:1.0.0
+              imagePullPolicy: Always
```
Comment (joshulyne) [inline deploy/kuberhealthy-prometheus-operator.yaml]: Set to IfNotPresent

File: deploy/kuberhealthy-prometheus.yaml [hunk 5]
```diff
@@ -1,15 +1,46 @@
...
+  group: comcast.github.io
+  version: v1
+  scope: Namespaced
+  names:
+    plural: khchecks
+    singular: khcheck
+    kind: KuberhealthyCheck
+    shortNames:
+      - khc  
+
```
Comment (joshulyne) [inline deploy/kuberhealthy-prometheus.yaml]: empty line in yaml?

File: deploy/kuberhealthy-prometheus.yaml [hunk 6]
```diff
@@ -313,11 +373,32 @@ spec:
...
+spec:
+  schedule: "*/3 * * * *"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+            - name: check-reaper
+              image: quay.io/comcast/check-reaper:1.0.0
+              imagePullPolicy: Always
```
Comment (joshulyne) [inline deploy/kuberhealthy-prometheus.yaml]: set to IfNotPresent

File: deploy/kuberhealthy.yaml [hunk 7]
```diff
@@ -280,11 +349,32 @@ spec:
...
+spec:
+  schedule: "*/3 * * * *"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+            - name: check-reaper
+              image: quay.io/comcast/check-reaper:1.0.0
+              imagePullPolicy: Always
```
Comment (joshulyne) [inline deploy/kuberhealthy.yaml]: IfNotPresent

File: cmd/check-reaper/check-reaper.yaml (hunk 1 above)
Comment (integrii) [reply cmd/check-reaper/check-reaper.yaml]: ✅

**Pull request merged**